tools/util/injectorutil.go

```diff
@@ -621,6 +675,33 @@ func createSignedStake(
 	return selp, createStake, nil
 }
 
+func injectAction(
+	wg *sync.WaitGroup,
+	selp *action.SealedEnvelope,
+	c iotexapi.APIServiceClient,
+	retryNum int,
+	retryInterval int,
+	pendingActionMap *ttl.Cache,
+	callback func(),
+) {
+	bo := backoff.WithMaxRetries(backoff.NewConstantBackOff(time.Duration(retryInterval)*time.Second), uint64(retryNum))
+	if err := backoff.Retry(func() error {
+		_, err := c.SendAction(context.Background(), &iotexapi.SendActionRequest{Action: selp.Proto()})
+		return err
+	}, bo); err != nil {
+		log.L().Error("Failed to inject action", zap.Error(err))
```

Comment:
if injection failed, shall we quit here or continue to add it to pending action map?

Reply:
yeah, should not add to pending map, fixed